druid/src/paintbrush.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Painting the background of a widget.
+
+use crate::piet::{FixedGradient, LinearGradient, RadialGradient};
+use crate::widget::Painter;
+use crate::{Color, Env, Key, PaintBrush, PaintCtx, Rect, RenderContext, Widget};
+
+/// Something that can be used as the background for a widget.
+///
+/// This represents anything that can be painted inside a widgets [`paint`]
+/// method; that is, it may have access to the [`Data`] and the [`Env`].
+///
+/// [`paint`]: widget/trait.Widget.html#tymethod.paint
+/// [`Data`]: trait.Data.html
+/// [`Env`]: struct.Env.html
+pub enum BackgroundBrush<T> {
+    Color(Color),
+    ColorKey(Key<Color>),
+    Linear(LinearGradient),
+    Radial(RadialGradient),
+    Fixed(FixedGradient),
```

Comment:
I guess that we might have `LinearKey`, `RadialKey` and `FixedKey` in the future without breaking changes.

Reply:
possibly; I'm not 100% sure those types belong in the env. But good catch, I will mark this as `non_exhaustive` to ensure adding an item is non-breaking.